internal/subshell/subshell.go

```diff
@@ -182,20 +183,24 @@ func DetectShell(cfg sscommon.Configurable) (string, string) {
 		}
 	}()
 
-	binary = os.Getenv("SHELL")
-	if binary == "" && runtime.GOOS == "windows" {
-		binary = detectShellWindows()
+	if os.Getenv(constants.OverrideShellEnvVarName) != "" {
+		binary = os.Getenv(constants.OverrideShellEnvVarName)
+	}
+
+	if binary == "" {
+		binary = detectShellParent()
 	}
 
 	if binary == "" {
 		binary = configured
 	}
+
 	if binary == "" {
-		if runtime.GOOS == "windows" {
-			binary = "cmd.exe"
-		} else {
-			binary = "bash"
-		}
+		binary = os.Getenv(SHELL_ENV_VAR)
+	}
+
+	if binary == "" {
+		binary = OS_DEFULAT
 	}
```

Comment:
Can we change this into a switch case? We don't need to assert `binary == nil`, instead assert whether we have something to populate it with.

Reply:
We can however this leads to us calling functions like `Getenv` or `detectShellParent()` twice. We could also move the function calls outside of the switch but then there will be cases where we are calling them needlessly. 

I'm not sure how much of a performance impact `detectShellParent()` has but we would be adding inefficiency for a marginal gain in readability. I'm fine with whatever direction we want to go but I wanted to flag this before I make the change.